Seniv#10 ShoppingCartTest

GrishaSeniv opened the pull request:
The first test adds the first item to the cart and checks by name. The second test removes this item and checks the cart.

GrishaSeniv committed:
Добавив EditAccountTest

GrishaSeniv committed:
Remove extra space and update a locator

GrishaSeniv committed:
The first test adds the first item to the cart and checks by name. The second test removes this item and checks the cart.

adomanich commented on src/main/java/com/opencart/steps/AddToShoppingCartBL.java:
```diff
@@ -0,0 +1,22 @@
+import com.opencart.pages.AddToShoppingCartPage;
+import org.openqa.selenium.WebElement;
+
+public class AddToShoppingCartBL {
+    private AddToShoppingCartPage addToShoppingCartPage;
+
+    public AddToShoppingCartBL() {
+        addToShoppingCartPage = new AddToShoppingCartPage();
+    }
+
+    public AddToShoppingCartBL clickOnAddToShoppingCartButton() {
+        addToShoppingCartPage.getAddToShoppingCartButton().get(0).click();
+        return this;
+    }
+
+    public ShoppingCartBL clickOnShoppingCartButton() {
+        addToShoppingCartPage.getShoppingCartButton().click();
+        return new ShoppingCartBL();
+    }
+}
```
I think you can move it HeaderPageBL business logic because you can get access to the shopping cart button from any page

adomanich commented on src/main/java/com/opencart/pages/EditAccountPage.java:
```diff
@@ -0,0 +1,14 @@
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class EditAccountPage extends BasePage {
```
Could you, please, rename it to LeftSideMenuOptionsPage or something like that? because it is not only related to EditAccount, there are also such options like WishList, Password, etc.

adomanich replied:
Please, make changes according to this comment

TarasHrysiukk commented on src/main/java/com/opencart/pages/AddToShoppingCartPage.java:
```diff
@@ -0,0 +1,22 @@
+
+public class AddToShoppingCartPage extends BasePage {
+    @FindBy(xpath = "//div[@class='product-thumb transition']//*[@class='button-group']/*[@type='button'][1]")
+    private List<WebElement> addToShoppingCartButton;
```
що робить цей рядок?

GrishaSeniv committed:
1)Add a new functionality and Tests(1.Add product to shop cart with additional optionals; 2.Remove this product from shop cart with additional optionals)
2)Move from AddToShoppingCartBL Business Logic to HeaderPageBL
3)In DriverUtils.java specifies the path to the files to be uploaded to the site. You must create the same file.

adomanich commented on src/main/java/com/opencart/steps/ShoppingCartBL.java:
```diff
@@ -0,0 +1,29 @@
+        return this;
+    }
+
+    public ShoppingCartBL verifyRemoveButton() {
```
Could you please rename it to verifyThatShoppingCartIsEmpty?

adomanich commented on src/main/java/com/opencart/util/DriverUtils.java:
```diff
@@ -17,6 +17,10 @@ public void clickOnElementJS(WebElement webElement) {
     }
 
+    public void uploadFileElementJs(WebElement webElement) {
+        getJSExecutor().executeScript("arguments[0].value='C:\\test.jpg'", webElement);
```
Please, put the file path as a method parameter

GrishaSeniv committed:
1)put the file path as a method parameter
2)rename to verifyThatShoppingCartIsEmpty

adomanich commented on src/main/java/com/opencart/steps/HeaderPageBL.java:
```diff
@@ -19,4 +28,66 @@ public RegisterPageBL clickOnRegisterButton() {
+        addToShoppingCartPage.getGoToShoppingCartStore().click();
+        return new ShoppingCartBL();
+    }
+
```
Remove extra space

adomanich commented on src/main/java/com/opencart/steps/HeaderPageBL.java:
```diff
@@ -19,4 +28,66 @@ public RegisterPageBL clickOnRegisterButton() {
+        AddToShoppingCartModel addToShoppingCartModel = AddToShoppingCartModelRepository.getAddToShoppingCartModel();
+        inputText(addToShoppingCartModel.getTextInput());
+        inputAreaText(addToShoppingCartModel.getTextAreaInput());
+        radioButton();
+        checkboxButton();
+        sliderButton();
+        uploadFileButton();
+        addToCartButton();
```
Could you please, specify actions you perform in each method name? For example, clickOnRadioButton

GrishaSeniv committed:
1)Removed extra space
2)Renamed EditAccountPage to LeftSideMenuOptionsPage
3)Renamed some method names in HeaderPageBL

GrishaSeniv committed:
1)Added a new test verifyUpdateButtonTest
2)Renamed ShoppingCartVerifyProductByName to ShoppingCartVerification

adomanich commented on src/main/java/com/opencart/pages/ProductPage.java:
```diff
@@ -0,0 +1,15 @@
+
+import java.util.List;
+
+public class ProductPage extends BasePage {
+    @FindBy(xpath = "//div[@class='product-thumb transition']//*[@class='button-group']/*[@type='button'][1]")
+    private List<WebElement> productNameButton;
+
+    public List<WebElement> getProductNameButton() {
+        return productNameButton;
+    }
+}
```
Could you please describe, why do you need ProductPage and ProductPageBL?
